consensus/istanbul/backend/handler.go

```diff
@@ -133,6 +133,19 @@ func (sb *Backend) handleConsensusMsg(peer consensus.Peer, payload []byte) error
 			return nil
 		}
 
+		msg := new(istanbul.Message)
+
+		// Verify that this message is coming from a legitimate validator before forwarding.
+		checkValidatorSignature := func(data []byte, sig []byte) (common.Address, error) {
+			block := sb.currentBlock()
+			valSet := sb.getValidators(block.Number().Uint64(), block.Hash())
+			return istanbul.CheckValidatorSignature(valSet, data, sig)
+		}
```

Comment:
For reviewer: this is functionally a copy of https://github.com/celo-org/celo-blockchain/blob/master/consensus/istanbul/core/core.go#L640 , which is private.
Does it make sense to copy or is it safe to expose `checkValidatorSignature` on `core.go`?

Reply:
This looks good to me!